controllers/kserve_customcacert_controller.go

```diff
@@ -0,0 +1,203 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/opendatahub-io/odh-model-controller/controllers/constants"
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	kserveCustomCACertConfigMapName = constants.KServeCACertConfigMapName
+	odhGlobalCACertConfigMapName    = constants.ODHGlobalCertConfigMapName
+)
+
+type KServeCustomCACertReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Log    logr.Logger
+}
+
+// reconcileConfigMap watch odh global ca cert and it will create/update/delete kserve custom cert configmap
+func (r *KServeCustomCACertReconciler) reconcileConfigMap(configmap *corev1.ConfigMap,
+	ctx context.Context) error {
+	// Initialize logger format
+	log := r.Log
+
+	odhCustomCertData := configmap.Data[constants.ODHCustomCACertFileName]
+	if odhCustomCertData != "" {
```

Comment:
Can this if chain be flattened? This looks pretty convoluted

Reply:
I agree. I made it more clear.

Reply:
@dtrifiro could you review it again?

Reply:
Source is refactoryed so I will close this comemnt